client/components/post-schedule/utils.js

```diff
@@ -1,9 +1,32 @@
 /**
  * External dependencies
  */
+import { endsWith, includes } from 'lodash';
 import { moment } from 'i18n-calypso';
 
 /**
+ * Checks whether the passed time  format is of a 12-hour format.
+ *
+ * @see https://wikipedia.org/wiki/12-hour_clock
+ *
+ * @param  {String}  timeFormat Time format.
+ * @return {Boolean}            Whether it's a 12-hour time format.
+ */
+const is12hr = ( timeFormat ) => (
+	// 12-hour format of an hour without leading zeros.
+	includes( timeFormat, 'g' ) ||
+
+	// 12-hour format of an hour with leading zeros.
+	includes( timeFormat, 'h' ) ||
+
+	// Lowercase Ante meridiem and Post meridiem.
+	endsWith( timeFormat, 'a' ) ||
+
+	// Uppercase Ante meridiem and Post meridiem.
+	endsWith( timeFormat, 'A' )
```

Comment:
Optional, but this can probably be shortened to:
```javascript
/[gh]|[aA]$/
```

👍  The comments are still super helpful though